### PR TITLE
Task 2.1 - added Alcohol class and overrided clone() method

### LINKED ISSUE #40: Task 2.1 - class Alcohol (methods)
W klasie Alcohol dopisz brakujące metody oraz ich implementacje.

```
// override from Cargo
size_t getPrice() const override;
std::string getName() const override { return name_; }
size_t getAmount() const override { return amount_; }
size_t getBasePrice() const override { return base_price_; }
Cargo& operator+=(size_t amount) override;
Cargo& operator-=(size_t amount) override;
bool operator==(Cargo& cargo) const override;
```

### PR DESCRIPTION
**Pull request by Petrus9107**
Left the spoil mechanism for alcohol class as well, regarding alcohol when time increases its price the getPrice() method should consider it and the spoil mechanism may require change as well.

### CODE REVIEW

**Review by mariuszlisowski — APPROVED** [review]

File: shm/inc/alcohol.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            : Cargo(name, amount, basePrice)
+            , timeToSpoil_(timeToSpoil)
+    {}
+    
+    Alcohol& operator--();
+    Alcohol& operator--(int);
+    bool operator==(const Cargo& alcohol) const;
+    bool operator!=(const Cargo& alcohol) const;
+    Cargo& operator+=(size_t amount);
+    Cargo& operator-=(size_t amount);
```
Comment (m-kop) [inline shm/inc/alcohol.hpp]: override

File: shm/inc/alcohol.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    size_t getPrice() const override;
+    std::string getName() const override;
+    size_t getAmount() const override;
+    size_t getBasePrice() const override;
+    size_t getTimeToSpoil() const;
+    size_t getTimeToSpoilLeft() const;
+
+    Cargo* clone() override { return new Alcohol( *this);}
```
Comment (m-kop) [inline shm/inc/alcohol.hpp]: why not use shared_ptr here, where this memory is deleted?

File: shm/inc/fruit.hpp [hunk 3]
```diff
@@ -25,6 +25,8 @@ class Fruit : public Cargo {
     size_t getTimeToSpoil() const;
     size_t getTimeToSpoilLeft() const;
 
+    Cargo* clone() override { return new Fruit( *this);}
```
Comment (m-kop) [inline shm/inc/fruit.hpp]: ditto

File: shm/inc/item.hpp [hunk 4]
```diff
@@ -31,6 +31,8 @@ class Item : public Cargo {
     size_t getBasePrice() const override;
     Rarity getRarity() const;
 
+    Cargo* clone() override { return new Item( *this);}
```
Comment (m-kop) [inline shm/inc/item.hpp]: ditto

**Review by m-kop — CHANGES_REQUESTED** [review]

File: shm/inc/alcohol.hpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#include <memory>
+
+#include "cargo.hpp"
+
+class Alcohol : public Cargo {
+public:
+    Alcohol(const std::string& name, size_t amount, size_t basePrice, size_t timeToSpoil)
+            : Cargo(name, amount, basePrice)
+            , timeToSpoil_(timeToSpoil)
+    {}
```
Comment (mariuszlisowski) [inline shm/inc/alcohol.hpp]: Cargo constructor needs Time*.

Alcohol(const std::string& name,
            size_t amount,
            size_t basePrice,
            size_t timeToSpoil,
            Time* time)
        : Cargo(name, amount, basePrice, time), timeToSpoil_(timeToSpoil)
    {}

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Pull request merged**